gwpy/frequencyseries/_fdcommon.py

```diff
@@ -50,15 +50,21 @@ def fdfilter(data, *filt, **kwargs):
 
     # parse filter
     if fs is None:
-        fs = 2 * (data.shape[-1] * data.df).to('Hz').value
```

Comment:
I think technically the old version is actually off by 2 * delta f. Data.shape[-1] gives len L.
e.g. freqs = [0, 1, 2..99], 2 * 100 * 1 = 200. Whereas it should be 2 * 99  = 198?

Reply:
This depends heavily how the frequency series was generated. In general, we cannot guarantee what `fs` is given an arbitrary array input, but using `frequencies[-1]` is probably the easiest to document and use.